File: test py/test_transfer_api.py
#!/usr/bin/env python
"""
Simple test script for transfer API endpoints.
"""

import os
import sys
import django

# Setup Django first
os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'peykan.settings')
django.setup()

from django.test import TestCase
from django.urls import reverse
from rest_framework.test import APIClient
from rest_framework import status
from transfers.models import TransferRoute, TransferRoutePricing, TransferOption

def test_transfer_api():
    """Test transfer API endpoints."""
    print("=" * 80)
    print("TRANSFER API TEST")
    print("=" * 80)
    
    # Create API client
    client = APIClient()
    
    # Test 1: Get all routes
    print("\n1. Testing GET /api/transfers/routes/")
    response = client.get('/api/transfers/routes/')
    print(f"Status: {response.status_code}")
    if response.status_code == 200:
        data = response.json()
        print(f"Found {len(data)} routes")
        for route in data[:3]:  # Show first 3 routes
            print(f"  - {route.get('origin')} → {route.get('destination')}")
    else:
        print(f"Error: {response.text}")
    
    # Test 2: Get specific route
    print("\n2. Testing GET /api/transfers/routes/{slug}/")
    routes = TransferRoute.objects.all()
    if routes.exists():
        route = routes.first()
        response = client.get(f'/api/transfers/routes/{route.slug}/')
        print(f"Status: {response.status_code}")
        if response.status_code == 200:
            data = response.json()
            print(f"Route: {data.get('origin')} → {data.get('destination')}")
            print(f"Pricing options: {len(data.get('pricing', []))}")
        else:
            print(f"Error: {response.text}")
    
    # Test 3: Get pricing for a route
    print("\n3. Testing GET /api/transfers/routes/{slug}/pricing/")
    if routes.exists():
        route = routes.first()
        response = client.get(f'/api/transfers/routes/{route.slug}/pricing/')
        print(f"Status: {response.status_code}")
        if response.status_code == 200:
            data = response.json()
            print(f"Found {len(data)} pricing options")
            for pricing in data[:3]:  # Show first 3 pricing options
                print(f"  - {pricing.get('vehicle_type')}: ${pricing.get('base_price')}")
        else:
            print(f"Error: {response.text}")
    
    # Test 4: Get transfer options
    print("\n4. Testing GET /api/transfers/options/")
    response = client.get('/api/transfers/options/')
    print(f"Status: {response.status_code}")
    if response.status_code == 200:
        data = response.json()
        print(f"Found {len(data)} options")
        for option in data:
            print(f"  - {option.get('option_type')}: ${option.get('price')}")
    else:
        print(f"Error: {response.text}")
    
    # Test 5: Calculate pricing
    print("\n5. Testing POST /api/transfers/calculate-price/")
    if routes.exists():
        route = routes.first()
        pricing = TransferRoutePricing.objects.filter(route=route).first()
        if pricing:
            payload = {
                'route_slug': route.slug,
                'vehicle_type': pricing.vehicle_type,
                'passenger_count': 2,
                'luggage_count': 2,
                'trip_type': 'one_way',
                'outbound_date': '2024-12-25',
                'outbound_time': '14:00',
                'selected_options': []
            }
            response = client.post('/api/transfers/calculate-price/', payload, format='json')
            print(f"Status: {response.status_code}")
            if response.status_code == 200:
                data = response.json()
                print(f"Base price: ${data.get('base_price')}")
                print(f"Final price: ${data.get('final_price')}")
            else:
                print(f"Error: {response.text}")
    
    print("\n" + "=" * 80)
    print("TRANSFER API TEST COMPLETED")
    print("=" * 80)

if __name__ == '__main__':
    test_transfer_api() 

File: md/test_transfer_api.js
// Test script for Transfer API
const axios = require('axios');

const API_BASE_URL = 'http://localhost:8000/api/v1';

async function testTransferAPI() {
  try {
    console.log('Testing Transfer API...');
    
    // Test 1: Get all transfer routes
    console.log('\n1. Testing GET /transfers/routes/');
    const routesResponse = await axios.get(`${API_BASE_URL}/transfers/routes/`);
    console.log('Status:', routesResponse.status);
    console.log('Routes count:', routesResponse.data.count);
    console.log('Routes:', routesResponse.data.results.map(r => `${r.origin} -> ${r.destination}`));
    
    // Test 2: Get a specific route
    if (routesResponse.data.results.length > 0) {
      const routeId = routesResponse.data.results[0].id;
      console.log(`\n2. Testing GET /transfers/routes/${routeId}/`);
      const routeResponse = await axios.get(`${API_BASE_URL}/transfers/routes/${routeId}/`);
      console.log('Status:', routeResponse.status);
      console.log('Route details:', {
        name: routeResponse.data.name,
        origin: routeResponse.data.origin,
        destination: routeResponse.data.destination,
        pricing_summary: Object.keys(routeResponse.data.pricing_summary || {})
      });
    }
    
    console.log('\n✅ Transfer API is working correctly!');
    
  } catch (error) {
    console.error('❌ Error testing Transfer API:', error.message);
    if (error.response) {
      console.error('Response status:', error.response.status);
      console.error('Response data:', error.response.data);
    }
  }
}

testTransferAPI(); 